doc/employees.md

```diff
@@ -1 +1,605 @@
 # Employees Module
+Module | Description 
+--- | --- | --- | ---
+[Add](#add) | POST | Add New Data Employee
+[View](#view) | GET | View Data Employee 
+[Edit](#edit) | PUT | Edit Data Employee 
+[Delete](#delete) | DELETE | Delete Data Employee
+
+## <a name="add"></a>Add
+
+### Endpoint 
+POST /employees
+
+### Database
+![](./employees.png)
+
+for add, you need login with company_id from companies tables and branch_id from branches tables
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | *
+Accept | application/json
+Email | alamat@emailcdc.com
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+employee_kode | string | 20201201  
+employee_nama | string | ferry
+stall_code | string | Heavy Repair
+```
+{
+    "employee_kode": "20200301",
+    "employee_nama": "ferry",
+    "stall_code": "Heavy Repair"
+}
+```
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+500 | Internal Server Error | some un-handle error in server 
+201 | Created | Created
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-201",
+    "status_message": "Created",
+    "data": {
+        "id":"5fbf4995-b4ac-4a2e-9cad-4021cf944ef7",
+        "employee_kode": "20200301",
+        "employee_nama": "ferry",
+        "stall_code": "Heavy Repair",
+        "jabatan_id": "5d2c04c9-fffc-4175-ac06-148dcf944ef7",
+        "company_id": "5e1befd2-ab04-49c2-addd-5b25c0a8141e",
+        "created" : "2020-12-03 09:44:02",
+        "modified" : "2020-12-03 09:44:02"
+    }
+}
+```
+
+### Logic
+
+#### Validation
+- employee_kode : required and not empty
+- employee_nama : required and not empty
+- stall_code : required and not empty
+
```

Comment:
- jabatan_id : required, not empty and exist in jabatans table
- branch_id: if user login is main dealer user, branch_id is required, not empty, exist in branches table, and owned by user login.